File: Models/DTOs/OrderItemDTO.cs
using Swashbuckle.AspNetCore.Annotations;

namespace OrderAPI.Models.DTOs
{
    public class OrderItemDTO
    {
        [SwaggerSchema("Product's unique identifier")]
        public Guid ProductId { get; set; }

        [SwaggerSchema("Name of the product")]
        public string ProductName { get; set; } = string.Empty;

        [SwaggerSchema("Quantity of the product ordered")]
        public int Quantity { get; set; }

        [SwaggerSchema("Unit price of the product")]
        public decimal UnitPrice { get; set; }

        [SwaggerSchema("Total price for the item (Quantity * UnitPrice)", ReadOnly = true)]
        public decimal TotalPrice => Quantity * UnitPrice;
    }

}

File: models/dtos/order_item_dto.py
import uuid
from dataclasses import dataclass, field
from typing import Optional

@dataclass
class OrderItemDTO:

    product_id: uuid.UUID

    quantity: int

    product_name: Optional[str] = field(default=None)

    unit_price: Optional[float] = field(default=None)

    @property
    def total_price(self) -> Optional[float]:
        if self.unit_price is None:
            return None
        return self.quantity * self.unit_price
